.github/workflows/on_push_docs.yml

```diff
@@ -8,22 +8,88 @@ concurrency:
   group: on-push-docs
   cancel-in-progress: true
 
+permissions:
+  contents: "read"
+  id-token: "write"
+
 defaults:
   run:
     shell: bash
 
 jobs:
+  # Get latest release version from crates.io
+  # This excludes any prerelease builds, e.g. `0.15.0-alpha.1` or `rc` or similar.
+  # We get it from crates.io because it's the strongest indicator of the latest
+  # fully released version we have available, and there is no better way to retrieve
+  # that in the context of this branch, because we don't want to rely on the contents
+  # of a local `Cargo.toml` or the git branch name.
+  get-version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.versioning.outputs.crate_version }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: prefix-dev/setup-pixi@v0.5.2
+        with:
+          pixi-version: v0.19.0
+
+      - name: Get version
+        id: versioning
+        run: |
+          crate_version=$(pixi run python scripts/ci/crates.py get-version --from=cratesio --skip-prerelease)
+          echo "crate_version=$crate_version" >> "$GITHUB_OUTPUT"
+
+  build-search-index:
+    runs-on: ubuntu-latest-16-cores
+    needs: [get-version]
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: ./.github/actions/setup-rust
+        with:
+          cache_key: "build-linux"
+          save_cache: false
+          workload_identity_provider: ${{ secrets.GOOGLE_WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ secrets.GOOGLE_SERVICE_ACCOUNT }}
+          toolchains: "nightly"
+
+      - uses: prefix-dev/setup-pixi@v0.5.2
+        with:
+          pixi-version: v0.19.0
+
+      - name: Install rerun-sdk
+        run: |
+
+      - name: Build search index
+        env:
+          RUSTFLAGS: "-Awarnings --cfg disable_web_viewer_server"
```

Comment:
On top of disabling the web viewer server here, warnings are also disabled, because `nightly` toolchain has a different set of warnings enabled by default. We already test with `-D warnings` elsewhere, and the output is _really_ noisy with the nightly warnings. I'm not sure if we should test with `-D warnings` on nightly somewhere else, because none of us actually run that locally so it'd be a new source of friction.